test/async-observer.test.js

```diff
@@ -324,6 +324,86 @@ describe('async observer', function() {
       }
     );
   });
+
+  it('should call after operation hook on error', function(done) {
+    const context = {
+      req: {},
+    };
+    const operationError = new Error('The operation failed without result');
+    let passesBy = 0;
+
+    function fn(context, done) {
```

Comment:
Let's use a more descriptive name please, e.g.

```suggestion
    function failingOperation(context, done) {
```

or simply:

```suggestion
    function fail(context, done) {
```

The same comment applies to other newly added tests below too.